biogeochem/EDCanopyStructureMod.F90

```diff
@@ -307,9 +308,19 @@ subroutine canopy_structure( currentSite , bc_in )
        enddo ! do while(area_not_balanced)
 
 
-       ! Set current canopy layer occupancy indicator.
-       currentPatch%NCL_p = min(nclmax,z)
-
+       ! Save number of canopy layers to the patch structure
+       
+       if(z > nclmax) then
+          write(fates_log(),*) 'FATES generated more canopy layers than scratch-space allows'
+          write(fates_log(),*) 'Predicted: ',z
+          write(fates_log(),*) 'Scratch space (nclmax, see EDParamsMod.F90): ',nclmax
+          write(fates_log(),*) 'Consider increasing nclmax if this value is to low'
+          write(fates_log(),*) 'and you think this number of canopy layers is reasonable.'
+          call endrun(msg=errMsg(sourcefile, __LINE__))
+       else
+          currentPatch%NCL_p = z
+       end if
+       
```

Comment:
I'm open to allowing termination here. I suppose it will allow the model to continue working when people are testing strange edge-case parameter combinations that generate more than 5 canopy layers.

Reply:
@mpaiao , I looked at this again. The calls to termination that precede this section should be ensuring that z <= nclmax.  If z is larger than nclmax it should be in error at this point.